### PR TITLE
timers and ACKs

### LINKED ISSUE #13: DTLS 1.3: What should the retransmission timer recommendation be?
RFC 7925 RECOMMENDs a very high (9s) initial RTO for DTLS 1.2 handshake

Reasons:

1. to avoid spurious retransmits when the public key operation is taking long on a constrained node
1. to avoid congestion on LLNs when the loss is genuine and the flight to retransmit is bulky (e.g., certificate) and maybe fragmented
1. to cater for very high latency variance in certain access technology (e.g., GSM-SMS)

DTLS 1.3 now offers per-record retransmission and therefore less congestion risk on genuine loss, which makes 2. less relevant. 

The question is: should we soften the initial retransmission timers recommendations for 1.3?

Options:

* Keep RFC 7925 — 9s
* Relax to RFC 2988 — 3s
* Further relax to RFC 6347, RFC-to-be 9147 (both inherit from RFC 6298) — 1s

### LINKED ISSUE #18: RRC timer considerations
RRC defines a timer value associated with the path validation algorithm with a default value of 1s.  This may not be OK in certain  IoT settings - for example, observed RTTs in NB-IoT are ~2.5s.
We need to work out whether there is any guidance we can provide in this area.

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #13
Fix #18

### CODE REVIEW

File: draft-ietf-uta-tls13-iot-profile.md [hunk 1]
```diff
@@ -200,11 +208,27 @@ negotiated independently.
...
+In general, the recommendations in {{Section 7.3 of DTLS13}} regarding ACKs apply.
+In particular, "[w]hen DTLS 1.3 is used in deployments with lossy networks, such as low-power, long-range radio networks as well as low-power mesh networks, the use of ACKs is recommended" to signal any sign of disruption or lack of progress.
+This allows for selective or early retransmission, which leads to more efficient use of bandwidth and memory resources.
+
+Due to the vast range of network technologies used in IoT deployments, from wired LAN to GSM-SMS, it's not possible to provide a universal recommendation for an initial timeout.
+Therefore, it is RECOMMENDED that DTLS 1.3 implementations provide a way for their users to explicitly set the initial timer value.
+Users SHOULD set the initial timeout to be twice the expected RTT but no less than 1000 ms.
+A sub-second initial timeout MAY be set for specific application/network combinations
+If no RTT estimates are available, 1 second is a good recommendation for the general internet.
+
```
Comment (thomas-fossati) [inline draft-ietf-uta-tls13-iot-profile.md:224]:
```suggestion
```

**Comment by thomas-fossati** [comment]
cc @julek-wolfssl @boaks

File: draft-ietf-uta-tls13-iot-profile.md [hunk 2]
```diff
@@ -200,11 +208,22 @@ negotiated independently.
 
 The discussion in Section 10 of {{!RFC7925}} is applicable.
 
-# Timeouts
+# Timers and ACKs
+
+Compared to DTLS 1.2 timeout-based whole flight retransmission, DTLS 1.3 ACKs sensibly decrease the risk of congestion collapse which was the basis for the very conservative recommendations given in Section 11 of {{!RFC7925}}.
```
Comment (thomas-fossati) [inline draft-ietf-uta-tls13-iot-profile.md:213]:
```suggestion
Compared to DTLS 1.2 timeout-based whole flight retransmission, DTLS 1.3 ACKs sensibly decrease the risk of congestion collapse which was the basis for the very conservative recommendations given in {{Section 11 of !RFC7925}}.
```

File: draft-ietf-uta-tls13-iot-profile.md [hunk 3]
```diff
@@ -200,11 +208,22 @@ negotiated independently.
...
+Therefore, it is RECOMMENDED that DTLS 1.3 implementations allow users to explicitly set the initial timer value.
+Users SHOULD set the initial timeout to be twice the expected round-trip time (RTT), but no less than 1000ms.
+For specific application/network combinations, a sub-second initial timeout MAY be set.
+In cases where no RTT estimates are available, a 1 second initial timeout is suitable for the general Internet.
 
-The recommendation in Section 11 of {{!RFC7925}} is applicable. In particular
-this document RECOMMENDED to use an initial timer value of 9 seconds with
-exponential back off up to no less then 60 seconds.
+For RRC, the recommendations in {{Section 7.5 of !I-D.ietf-tls-dtls-rrc}} apply.
+Just like the handshake initial timers, it is RECOMMENDED that DTLS 1.2 and 1.3 implementations offer an option for their users to explicitly set the RRC timer.
```
Comment (thomas-fossati) [inline draft-ietf-uta-tls13-iot-profile.md:226]:
```suggestion
Just like the handshake initial timers, it is RECOMMENDED that DTLS 1.2 and 1.3 implementations offer an option for their developers to explicitly set the RRC timer.
```

Comment (boaks) [reply draft-ietf-uta-tls13-iot-profile.md:226]: My caused by my limited English, why is developer used twice and user still once?

Comment (thomas-fossati) [reply draft-ietf-uta-tls13-iot-profile.md:226]: > My caused by my limited English, why is developer used twice and user still once?

I think "user" was consistently replaced by "developer" in what got eventually merged.

**Pull request merged**